src/main/java/org/truffleruby/core/regexp/RegexpNodes.java

```diff
@@ -125,7 +125,9 @@ public static DynamicObject matchCommon(RubyContext context, Node currentNode, R
 
         final Rope sourceRope = StringOperations.rope(string);
 
-        int match = context.getThreadManager().runUntilResult(currentNode, () -> matcher.searchInterruptible(startPos, range, Option.DEFAULT));
+        // Keep status as RUN because MRI has an uninterruptible Regexp engine
+        int match = context.getThreadManager().runUntilSuccessKeepRunStatus(currentNode,
```

Comment:
I'm curious how you found this issue. Is it something that can be spec'd?

Reply:
I was debugging a Thread#kill spec, and just adding a few `p :here` would put the Thread status to sleep as Symbol#inspect uses regexps. I checked MRI's source and there they don't change the status.
I also checked on MRI whether a very long Regexp match with a pathological Regexp would change the status but that can't be checked afaik as the GIL is held while doing the matching.
We could check it for us, at the price of executing a non-deterministic and long-running spec so to hope to see the Thread#status while it is doing the regexp match.

Reply:
I added a Truffle spec.